pkg/cmd/start/bootstrap.go

```diff
@@ -88,10 +91,61 @@ func bootstrap(ctx context.Context) manager.Manager {
 	setupControllers(ctx, mgr)
 	serveCRMetrics(ctx, cfg, namespace)
 	createMetricsService(ctx, cfg, namespace)
+	detectOAuthProxyImageStream(ctx, mgr)
 
 	return mgr
 }
 
+func detectOAuthProxyImageStream(ctx context.Context, mgr manager.Manager) {
+	tracer := global.TraceProvider().GetTracer(v1.BootstrapTracer)
+	ctx, span := tracer.Start(ctx, "detectOAuthProxyImageStream")
+	defer span.End()
+
+	if viper.GetString("platform") != v1.FlagPlatformOpenShift {
+		log.Debug("Not running on OpenShift, so won't configure OAuthProxy imagestream.")
+		return
+	}
+
+	imageStreamNamespace := viper.GetString("openshift-oauth-proxy-imagestream-ns")
+	imageStreamName := viper.GetString("openshift-oauth-proxy-imagestream-name")
+	if imageStreamNamespace == "" || imageStreamName == "" {
+		log.WithFields(log.Fields{
+			"namespace": imageStreamNamespace,
+			"name":      imageStreamName,
+		}).Info("OAuthProxy ImageStream namespace and/or name not defined")
+		return
+	}
+
+	imageStream := &osimagev1.ImageStream{}
+	namespacedName := types.NamespacedName{
+		Name:      imageStreamName,
+		Namespace: imageStreamNamespace,
+	}
+	if err := mgr.GetAPIReader().Get(ctx, namespacedName, imageStream); err != nil {
+		log.WithFields(log.Fields{
+			"namespace": imageStreamNamespace,
+			"name":      imageStreamName,
+		}).WithError(err).Error("Failed to obtain OAuthProxy ImageStream")
+		tracing.HandleError(err, span)
+		return
+	}
+
+	if len(imageStream.Status.Tags) == 0 {
+		log.WithFields(log.Fields{
+			"namespace": imageStreamNamespace,
+			"name":      imageStreamName,
+		}).Error("OAuthProxy ImageStream has no tags")
+		return
+	}
+
+	image := fmt.Sprintf("%s:%s", imageStream.Status.DockerImageRepository, imageStream.Status.Tags[0].Tag)
```

Comment:
Is it OK to assume that we will always use the first tag? I would say that the answer is yes, at least for now, but not sure.

Reply:
Yes, the [comment on the type](https://github.com/openshift/api/blob/master/image/v1/types.go#L295-L296) indicates that the first tag is the current one.

> Tags are a historical record of images associated with each tag. The first entry in the TagEvent array is the currently tagged image